serving/metadata-webhook/config/webhook.yaml

```diff
@@ -22,56 +22,52 @@ spec:
       affinity:
         podAntiAffinity:
           preferredDuringSchedulingIgnoredDuringExecution:
-          - podAffinityTerm:
-              labelSelector:
-                matchLabels:
-                  app: webhook
-              topologyKey: kubernetes.io/hostname
-            weight: 100
-
+            - podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    app: webhook
+                topologyKey: kubernetes.io/hostname
+              weight: 100
       serviceAccountName: controller
       containers:
-      - name: webhook
-        image: registry.ci.openshift.org/knative/serverless-metadata-webhook:knative-main
-        resources:
-          requests:
-            cpu: 20m
-            memory: 20Mi
-          limits:
-            cpu: 200m
-            memory: 200Mi
-        env:
-        - name: SYSTEM_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
-        - name: KUBERNETES_MIN_VERSION
-          value: v1.0.0
-        - name: CONFIG_LOGGING_NAME
-          value: config-logging
-        - name: METRICS_DOMAIN
-          value: knative.dev/samples
-        - name: WEBHOOK_NAME
-          value: webhook
-
-        securityContext:
-          allowPrivilegeEscalation: false
-          readOnlyRootFilesystem: true
-          runAsNonRoot: true
-          capabilities:
-            drop:
-            - all
-
-        readinessProbe: &probe
-          periodSeconds: 1
-          httpGet:
-            scheme: HTTPS
-            port: 8443
-            httpHeaders:
-            - name: k-kubelet-probe
-              value: "webhook"
-        livenessProbe: *probe
-
+        - name: webhook
+          image: quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-metadata-webhook:latest
```

Comment:
That's a fair point, @mgencur should we reconsider using a separate script for downstream and P/Z ?

Reply:
SO tests are using Quay, in particular, Konflux built images for testing as they are multi-arch and we don't need GH actions builds, that will also let us remove the CI Quay mirroring in the near future

https://github.com/openshift-knative/serverless-operator/blob/bd857dacc3f19686ea45e629ecf8a9b8e89979ac/test/lib.bash#L468-L500

https://github.com/openshift-knative/serverless-operator/blob/main/test/images-rekt.yaml

Reply:
ok, cool! The serverless-metadata-webhook is really a test image, so if that's the direction we're going with test images, that makes sense..

Reply:
@skonto This metadata-webhook is specific. It is similar to using images from midstream repositories (Serving, Eventing, etc.). We don't need it to be always up to date. It's enough if we use an image that was built after merge/push.
We can't use this approach in openshift-knative/serving itself because we need the test images to be built for tests in the same PR. So images built in CI by CI operator needs to be used there. It doesn't change.